### PR TITLE
Use gitignore patterns in stylelintignore; closes #1498

### LINKED ISSUE #1498: Use node-ignore for ignore patterns instead of globs
See discussion at https://github.com/stylelint/stylelint/issues/1452.

### PR DESCRIPTION
**Pull request by davidtheclark**
Ref #1498.

This took longer than anticipated, but I think it's done, or close, now. Please have a close look at the documentation changes to assess the implications of this change.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -92,8 +93,8 @@
     "lint:md": "remark . --quiet --frail",
     "lint": "npm-run-all --parallel lint:*",
     "pretest": "npm run lint",
-    "tape": "nyc babel-tape-runner \"src/**/__tests__/*.js\"",
-    "test": "npm run tape",
+    "tape": "babel-tape-runner \"src/**/__tests__/**/*.js\"",
```
Comment (jeddy3) [inline package.json]: 👍

**Comment by jeddy3** [comment]
Looks brilliant to me.

> Please have a close look at the documentation changes to assess the implications of this change.

I'm down with `.stylelintignore` and `ignorePath` providing an alternative method that behaves different to `ignoreFiles`.

Thanks for implementing this feature btw. It looked pretty gnarly.

**Comment by jeddy3** [comment]
@davidtheclark Oh no. I tried to rebase this so I could merge it in for you, but it's fouled up! Is this something you can quickly recover at your end, or shall I try to piece things back together at mine? Sorry about that.

**Comment by davidtheclark** [comment]
I still have the branch locally, so I will close this PR and submit another with the branch rebase on v7.

**Pull request closed without merging**